tests/RobotFramework/tests/cumulocity/supported_operations/mapper-publishing-agent-supported-ops.robot

```diff
@@ -0,0 +1,52 @@
+*** Settings ***
+Resource    ../../../../resources/common.resource
+Library    Cumulocity
+Library    ThinEdgeIO
+
+Test Tags    theme:c8y    theme:operation    theme:tedge-agent
+Test Setup    Custom Setup
+Test Teardown    Get Logs
+
+*** Test Cases ***
+Publish the tedge agent supported operations delete restart
+    # wait till list request is pushed out
+    Should Have MQTT Messages    tedge/commands/req/software/list
+    # stop mapper and remove the supported operations
+    Execute Command    sudo systemctl stop tedge-mapper-c8y
+    Execute Command    sudo rm -rf /etc/tedge/operations/c8y/*
+  
+    # the operation files must not exist
+    ThinEdgeIO.File Should Not Exist    /etc/tedge/operations/c8y/c8y_SoftwareUpdate
+    ThinEdgeIO.File Should Not Exist    /etc/tedge/operations/c8y/c8y_Restart
+    
+    # now restart the mapper
+    Execute Command    sudo systemctl start tedge-mapper-c8y
+    Should Have MQTT Messages    tedge/health/tedge-mapper-c8y
+    
+    # Check if the `c8y_SoftwareUpdate` and `c8y_Restart` ops files exists in `/etc/tedge/operations/c8y` directory
+    ThinEdgeIO.File Should Exist    /etc/tedge/operations/c8y/c8y_SoftwareUpdate
+    ThinEdgeIO.File Should Exist    /etc/tedge/operations/c8y/c8y_Restart
+
+    # Check if the tedge-agent supported operations exists in c8y cloud
+    Cumulocity.Should Contain Supported Operations    c8y_Restart    c8y_SoftwareUpdate   
+
+
+Agent gets the software list request once it comes up
+    Execute Command    sudo systemctl stop tedge-agent
+    # wait till health status down
+    Should Have MQTT Messages    tedge/health/tedge-agent
```

Comment:
I don't know if this API behaviour changed recently, but earlier this was checking the MQTT message trace right from the point where the device started up and will check for a message with the given topic anywhere in that trace. But, here I believe you wanted to check if the health status was generated after the `tedge-agent` was shutdown. For that, you need to pass the `date_from` parameter as well. You need to pass the `message_contains` or `message_pattern` args as well to make sure that the status value is `down`.

Reply:
Added `timestamp` to check the `specific` messages.